### PR TITLE
Refactor/nonstd to boost

### PR DESCRIPTION
**Pull request by x3medima17**
### Description of the Change
This PR entirely removes `nonstd` dependency from the project, and replaces `nonstd::optional` with `boost::optional`

### Benefits
One less dependency.

### Possible Drawbacks 
Many files to review.

### CODE REVIEW

File: iroha-cli/interactive/impl/interactive_query_cli.cpp [hunk 1]
```diff
@@ -249,13 +249,13 @@ namespace iroha_cli {
       auto res = handleParse<bool>(
           this, line, result_handlers_, result_params_descriptions_);
 
-      return not res.has_value() ? true : res.value();
+      return not res ? true : res.value();
```
Comment (Warchant) [inline iroha-cli/interactive/impl/interactive_query_cli.cpp]: return res.get_value_or(true)

File: iroha-cli/interactive/impl/interactive_transaction_cli.cpp [hunk 2]
```diff
@@ -460,14 +460,14 @@ namespace iroha_cli {
       // Find in result handler map
       auto res = handleParse<bool>(
           this, line, result_handlers_, result_params_descriptions);
-      return not res.has_value() ? true : res.value();
+      return not res ? true : res.value();
```
Comment (Warchant) [inline iroha-cli/interactive/impl/interactive_transaction_cli.cpp]: Same as above

File: irohad/model/converters/impl/json_block_factory.cpp [hunk 3]
```diff
@@ -70,17 +69,18 @@ namespace iroha {
...
-                    return nonstd::make_optional(transactions);
+                    return boost::make_optional(std::move(transactions));
                   };
             };
           };
           return std::accumulate(
               array.begin(),
               array.end(),
-              nonstd::make_optional<Block::TransactionsType>(),
+              boost::make_optional(Block::TransactionsType()),
```
Comment (Warchant) [inline irohad/model/converters/impl/json_block_factory.cpp]: make_optional is not parametrized?

File: irohad/model/generators/CMakeLists.txt [hunk 4]
```diff
@@ -25,7 +25,7 @@ add_library(model_generators
 
 target_link_libraries(model_generators
     model
-    optional
+    model
```
Comment (Warchant) [inline irohad/model/generators/CMakeLists.txt]: twice

File: irohad/model/converters/impl/json_block_factory.cpp (hunk 3 above)
Comment (x3medima17) [reply irohad/model/converters/impl/json_block_factory.cpp]: It can deduce type

File: iroha-cli/interactive/impl/interactive_status_cli.cpp [hunk 5]
```diff
@@ -130,18 +130,18 @@ namespace iroha_cli {
       auto res = handleParse<bool>(
           this, line, resultHandlers_, resultParamsDescriptions_);
 
-      return not res.has_value() ? true : res.value();
+      return not res ? true : res.value();
```
Comment (nickaleks) [inline iroha-cli/interactive/impl/interactive_status_cli.cpp]: Please rewrite it with get_value_or (as above)

File: iroha-cli/interactive/impl/interactive_transaction_cli.cpp [hunk 6]
```diff
@@ -334,7 +334,7 @@ namespace iroha_cli {
       auto val_int =
           parser::parseValue<boost::multiprecision::uint256_t>(params[2]);
       auto precision = parser::parseValue<uint32_t>(params[3]);
-      if (not val_int.has_value() || not precision.has_value()) {
+      if (not val_int || not precision) {
```
Comment (nickaleks) [inline iroha-cli/interactive/impl/interactive_transaction_cli.cpp]: please change to or

File: iroha-cli/interactive/impl/interactive_transaction_cli.cpp [hunk 7]
```diff
@@ -436,7 +436,7 @@ namespace iroha_cli {
       auto val_int =
           parser::parseValue<boost::multiprecision::uint256_t>(params[3]);
       auto precision = parser::parseValue<uint32_t>(params[4]);
-      if (not val_int.has_value() || not precision.has_value()) {
+      if (not val_int || not precision) {
```
Comment (nickaleks) [inline iroha-cli/interactive/impl/interactive_transaction_cli.cpp]: Please change to or

File: libs/crypto/keys_manager_impl.cpp [hunk 8]
```diff
@@ -118,37 +118,37 @@ namespace iroha {
     return true;
   }
 
-  nonstd::optional<keypair_t> KeysManagerImpl::loadKeys() {
+  boost::optional<keypair_t> KeysManagerImpl::loadKeys() {
     std::string pub_key;
     std::string priv_key;
 
     if (!loadFile(account_name_ + kPubExt, pub_key)
         || !loadFile(account_name_ + kPrivExt, priv_key))
```
Comment (nickaleks) [inline libs/crypto/keys_manager_impl.cpp]: Please change to or

File: libs/crypto/keys_manager_impl.cpp [hunk 9]
```diff
@@ -118,37 +118,37 @@ namespace iroha {
...
   }
 
-  nonstd::optional<keypair_t> KeysManagerImpl::loadKeys(
+  boost::optional<keypair_t> KeysManagerImpl::loadKeys(
       const std::string &pass_phrase) {
     std::string pub_key;
     std::string priv_key;
 
     if (!loadFile(account_name_ + kPubExt, pub_key)
         || !loadFile(account_name_ + kPrivExt, priv_key))
```
Comment (nickaleks) [inline libs/crypto/keys_manager_impl.cpp]: Please change to or

**Comment by Warchant** [comment]
It is easier to do changes from scratch :hushed:

File: irohad/model/converters/impl/json_block_factory.cpp [hunk 10]
```diff
@@ -70,17 +69,18 @@ namespace iroha {
...
               acc_transactions);
         };
-        return nonstd::make_optional<model::Block>()
+        model::Block bl;
```
Comment (Warchant) [inline irohad/model/converters/impl/json_block_factory.cpp]: is it used? why is this added?

File: libs/amount/amount.hpp [hunk 11]
```diff
@@ -30,6 +30,8 @@ namespace iroha {
    * operations on them
    */
   class Amount {
+    using uint256_t = boost::multiprecision::uint256_t;
```
Comment (Warchant) [inline libs/amount/amount.hpp]: why did you add this?

File: libs/common/CMakeLists.txt [hunk 12]
```diff
@@ -11,5 +11,4 @@ target_include_directories(common INTERFACE
     ${PROJECT_SOURCE_DIR}/libs/common
     )
 target_link_libraries(common INTERFACE
```
Comment (Warchant) [inline libs/common/CMakeLists.txt]: this can be removed
